File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;
import entity.Result;

import java.util.List;
import java.util.Map;

/**
 * 品牌 接口
 * @author wangyangkun
 * @date 2019/7/8 0008 19:36
 */
public interface BrandService {
    /**
     * 查询所有品牌
     * @return 品牌集合
     */
    List<TbBrand> findAll();

    /**
     * 分页查询品跑
     * @param currentPage 当前页
     * @param pageSize 每页显示条数
     * @return 结果对象
     */
    PageResult findPage(int currentPage, int pageSize);

    /**
     * 新增产品
     * @param brand 产品对象
     */
    void add(TbBrand brand);

    /**
     * 根据id查询产品
     * @param id id
     * @return 返回尸体对象
     */
    TbBrand findOne(Long id);

    /**
     * 修改品牌功能
     * @param brand 品牌对象
     */
    void update(TbBrand brand);

    /**
     * 根据id删除
     * @param ids 批量删除
     */
    void delete(Long[] ids);

    /**
     * 条件查询并分页
     * @param brand 查询实体条件
     * @param currentPage 当前页
     * @param pageSize 每页显示条数
     * @return 分页对象
     */
    PageResult findPage(TbBrand brand,int currentPage, int pageSize);

    /**
     * 查询所有品牌封装为map集合形式
     * @return
     */
    List<Map> selectOptionList();
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
//品牌服务层
app.service("brandService",function ($http) {
    //查询所有
    this.findAll=function () {
        return $http.get('../brand/findAll.do');
    };
    //分页查询
    this.findPage=function (currentPage,pageSize) {
        return $http.get('../brand/findByPage.do?currentPage='+currentPage +'&pageSize='+pageSize);
    };
    //添加品牌
    this.add=function (entity) {
        return $http.post("../brand/add.do",entity);
    };
    //修改品牌
    this.update=function (entity) {
        return $http.post("../brand/update.do",entity);
    };
    //按照id查询
    this.findOne=function (id) {
        return $http.get("../brand/findOne.do?id="+id);
    };
    //批量删除
    this.dele=function (ids) {
        return $http.get("../brand/delete.do?ids="+ids);
    };
    //按条件查询
    this.search=function (currentPage,pageSize,searchEntity) {
        return $http.post("../brand/search.do?currentPage="+currentPage+"&pageSize="+pageSize ,searchEntity);
    };
    //查询产品
    this.findBrandList=function () {
       return $http.get("../brand/findBrandList.do");
    }

});